### DISCUSSION TITLE
Laravel jetstream, How to override the ShareInertiaData class?

### DISCUSSION THREAD

**Discussion by murtazasultani** [laravel/framework#34882] [General]
I want to attach new data to the $page object. is there anyone to answer what is the best way to do this?
the class namespace: `Laravel\Jetstream\Http\Middleware;`

**Comment by Hmerman6006** [comment]
I do not know a best method, but if you set an array in the `HandleInertiaRequests` `share` function that has the same keys as the  `ShareInertiaData` array.  The `ShareInertiaData` array will be overwritten.
